From ccc8b442671ae85c16a36cc8c259bc4195e73b8f Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Mon, 3 Oct 2016 13:44:05 +0200
Subject: [PATCH 414/595] Bug 772171 - Editing quoted replies can break message
 content

Don't save the selection again in
e_editor_dom_fix_structure_after_delete_before_quoted_content() as later
the selection markers are not used to restore the selection, but they
would prevent the delete operation to process correctly.

Also restore the selection on other places when we are not using it
anymore.
---
 .../web-extension/e-editor-dom-functions.c         |  4 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/modules/webkit-editor/web-extension/e-editor-dom-functions.c b/modules/webkit-editor/web-extension/e-editor-dom-functions.c
index d00f330933..219f0ffecb 100644
--- a/modules/webkit-editor/web-extension/e-editor-dom-functions.c
+++ b/modules/webkit-editor/web-extension/e-editor-dom-functions.c
@@ -9127,6 +9127,7 @@ save_history_for_delete_or_backspace (EEditorPage *editor_page,
 			g_clear_object (&range_clone);
 			g_clear_object (&dom_selection);
 			g_warning ("History event was not saved for %s key", delete_key ? "Delete" : "Backspace");
+			e_editor_dom_selection_restore (editor_page);
 			return;
 		}
 
@@ -9352,6 +9353,8 @@ save_history_for_delete_or_backspace (EEditorPage *editor_page,
 
 	manager = e_editor_page_get_undo_redo_manager (editor_page);
 	e_editor_undo_redo_manager_insert_history_event (manager, ev);
+
+	e_editor_dom_selection_restore (editor_page);
 }
 
 gboolean
@@ -9405,7 +9408,6 @@ e_editor_dom_fix_structure_after_delete_before_quoted_content (EEditorPage *edit
 				e_editor_dom_selection_restore (editor_page);
 				save_history_for_delete_or_backspace (
 					editor_page, key_code == HTML_KEY_CODE_DELETE, control_key);
-				e_editor_dom_selection_save (editor_page);
 			}
 
 			/* Remove the empty block and move caret to the right place. */
-- 
2.15.0

